src/ngx_http_lua_module.c

```diff
@@ -738,6 +742,11 @@ ngx_http_lua_init(ngx_conf_t *cf)
 
     cln->data = lmcf;
     cln->handler = ngx_http_lua_sema_mm_cleanup;
+
+#if !(NGX_WIN32)
+    ngx_http_lua_pipe_init();
```

Comment:
Should this also be enclosed within `#ifdef HAVE_SOCKET_CLOEXEC_PATCH`? Otherwise there is no point initializing the pipe r/b/ tree if no pipe can be spawned.

I am starting to think that we should simplify our macro dependencies... E.g. come up with `HAVE_NGX_LUA_PIPE` defined by `ngx_http_lua_pipe.h` when `HAVE_SOCKET_CLOEXEC_PATCH` and `!(NGX_WIN32)` are met (and any other conditions, present or future). We then enclose those branches in existing init code paths within a single `#ifdef HAVE_NGX_LUA_PIPE`, which reads much cleaner.

Then, inside of `ngx_http_lua_pipe.c`, we have some extra work to do to ensure that we nop the exposed symbols when `#ifdef NGX_LUA_NO_FFI_API`. And we could enclose `ngx_http_lua_ffi_pipe_spawn` within `#ifdef HAVE_NGX_LUA_PIPE`, which also reads better.